LLK/scratch-www
# Upgrade to React 15

Also upgrade dependencies incompatible with React 15.

The styles in the upgraded react-modal broke our centering and added an an annoying blue border around the modal in Chrome, fix that.

Update search form submit button from an input[type=submit] to a button.  React 15 omits the value prop if it's blank, so the `<input type="submit" value="">` was rendered as `<input type="submit">`. This caused the input to display the default value of "Submit".  The button element is more appropriate for this behavior and doesn't have the missing value issue.

## Test cases
* Cross-browser tests:
  * Confirm modals look ok
  * Confirm sliders look ok
  * The homepage should load
  * You should be able to sign in and sign out
